eng/mark-shipped.ps1

```diff
@@ -0,0 +1,51 @@
+[CmdletBinding(PositionalBinding=$false)]
+param ()
+
+Set-StrictMode -version 2.0
+$ErrorActionPreference = "Stop"
+
+function MarkShipped([string]$dir) {
+    $shippedFilePath = Join-Path $dir "PublicAPI.Shipped.txt"
+    $shipped = Get-Content $shippedFilePath
+    if ($null -eq $shipped) {
+        $shipped = @()
+    }
+
+    $unshippedFilePath = Join-Path $dir "PublicAPI.Unshipped.txt"
+    $unshipped = Get-Content $unshippedFilePath
+    $removed = @()
+    $removedPrefix = "*REMOVED*";
+    Write-Host "Processing $dir"
+
+    foreach ($item in $unshipped) {
+        if ($item.Length -gt 0) {
+            if ($item.StartsWith($removedPrefix)) {
+                $item = $item.Substring($removedPrefix.Length)
+                $removed += $item
+            }
+            elseif ($item -ne "#nullable enable") {
+                $shipped += $item
+            }
+        }
+    }
+
+    $shipped | Sort-Object | Where-Object { -not $removed.Contains($_) } | Out-File $shippedFilePath -Encoding Ascii
+    "" | Out-File $unshippedFilePath -Encoding Ascii
```

Comment:
are those files really ascii and not utf8?

You could also do $_ -notin $removed for the condition.